src/devices/Charlieplex/CharlieplexSegment.cs

```diff
@@ -0,0 +1,244 @@
+        /// Initializes a new Charlieplex type that can be use for multiplex over a relatively small number of GPIO pins.
+        /// </summary>
+        /// <param name="pins">The set of pins to use.</param>
+        /// <param name="nodeCount">The count of nodes (like LEDs) that will be addressable. If 0, then the Charlieplex maximum is used for the pins provided (n^2-n).</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling.</param>
+        /// <param name="shouldDispose">True (the default) if the GPIO controller shall be disposed when disposing this instance.</param>
+        public CharlieplexSegment(int[] pins, int nodeCount = 0,  GpioController gpioController = null, bool shouldDispose = true)
+        {
+            if (pins.Length < 2)
+            {
+                throw new ArgumentException($"{nameof(CharlieplexSegment)}: 2 or more pins must be provided.");
+            }
+
+            var charlieCount = (int)Math.Pow(pins.Length, 2) - pins.Length;
+            if (nodeCount > charlieCount)
+            {
+                throw new ArgumentException($"{nameof(CharlieplexSegment)}: maximum count is {charlieCount} based on {pins.Length} pins. {nodeCount} was specified as the count.");
+            }
+
+            if (nodeCount == 0)
+            {
+                nodeCount = charlieCount;
+            }
+
+            if (gpioController is null)
+            {
+                gpioController = new GpioController();
+            }
+
+            // first two pins will be needed as Output.
+            gpioController.OpenPin(pins[0]);
+            gpioController.OpenPin(pins[1]);
+            gpioController.SetPinMode(pins[0], PinMode.Output);
+            gpioController.SetPinMode(pins[1], PinMode.Output);
+
+            for (int i = 2; i < pins.Length; i++)
+            {
+                gpioController.OpenPin(pins[i]);
+                gpioController.SetPinMode(pins[i], PinMode.Input);
+            }
+
+            _lastNode = new Node()
+            {
+                Anode = pins[1],
+                Cathode = pins[0]
+            };
+            _controller = gpioController;
+            _shouldDispose = shouldDispose;
+            _pins = pins;
+            _nodeCount = nodeCount;
+            _nodes = GetNodes(pins, nodeCount);
+        }
+
+        /// <summary>
+        /// The number of nodes (like LEDs) that can be addressed.
+        /// </summary>
+        public int NodeCount => _nodeCount;
+
+        /// <summary>
+        /// Write a PinValue to a node, to update Charlieplex segment.
+        /// Address scheme is 0-based. Given 8 nodes, addresses would be 0-7.
+        /// Displays nodes in their updated configuration for the specified duration.
+        /// </summary>
+        /// <param name="node">Node to update.</param>
+        /// <param name="value">Value to write.</param>
+        /// <param name="duration">Time to display segment, in milliseconds.</param>
```

Comment:
Why do we default to 10 ms? would it make sense to instead just have a Write which writes to the node the value but doesn't do it for a specific time? Basically the equivalent of what a gpioController.Write() would be.

Reply:
Good question. The underlying question is whether the LED should be lit as as a result of calling `.Write`.

Reply:
I thought more about this. Switched to default of 0 by default.